### PR TITLE
Delete replicaSet for app

### PR DESCRIPTION
**Pull request by linemos**

### CODE REVIEW

File: api/resourcedeleter.go [hunk 1]
```diff
@@ -76,6 +83,23 @@ func deleteDeployment(namespace string, deployName string, k8sClient kubernetes.
 	return "", nil
 }
 
+// replicaset should be deleted by foreground delete policy for deployment, temporary fix
+func deleteReplicaSet(namespace string, deployName string, k8sClient kubernetes.Interface) (result string, e error) {
+	appLabel := "app=" + deployName
+	replicaSets, err := k8sClient.AppsV1().ReplicaSets(namespace).List(k8smeta.ListOptions{LabelSelector: appLabel})
+	if err != nil {
+		return filterNotFound(fmt.Sprintf("could not delete replicaset: %s in namespace: %s: %s", deployName, namespace, err), err)
```
Comment (Kyrremann) [inline api/resourcedeleter.go]: kunne feilmeldingen heller vært at den ikke fant et replicaset? Siden den ikke prøver å slette den enda.

File: api/resourcedeleter.go [hunk 2]
```diff
@@ -76,6 +83,23 @@ func deleteDeployment(namespace string, deployName string, k8sClient kubernetes.
...
+	}
+
+	for _, replicaSet := range replicaSets.Items {
+		if err := k8sClient.AppsV1().ReplicaSets(namespace).Delete(replicaSet.Name, &k8smeta.DeleteOptions{}); err != nil {
+			return filterNotFound(fmt.Sprintf("could not delete replicaset: %s in namespace: %s: %s", deployName, namespace, err), err)
```
Comment (Kyrremann) [inline api/resourcedeleter.go]: Skulle den prøvd å slette de andre i listen, eller bare fastFail?

File: api/resourcedeleter_test.go [hunk 3]
```diff
@@ -37,7 +38,15 @@ func TestDeleteK8sResouces(t *testing.T) {
 	secretDef.ObjectMeta.ResourceVersion = resourceVersion
 	configMapDef := createConfigMapDef(AlertsConfigMapName, AlertsConfigMapNamespace, teamName)
 	configMapDef.ObjectMeta.ResourceVersion = resourceVersion
-	clientset := fake.NewSimpleClientset(serviceDef, deploymentDef, secretDef, configMapDef)
+	replicaSetDef := &k8sapps.ReplicaSet{
+		TypeMeta: k8smeta.TypeMeta{
+			Kind:       "ReplicaSet",
+			APIVersion: "v1",
```
Comment (Kyrremann) [inline api/resourcedeleter_test.go]: Kunne vi hentet `APIVersion` fra `k8sapps`?

**Comment by gtcno** [comment]
Why is this necessary? Does a  delete of a  deployment  not trigger a cascading delete of rs?

**Comment by linemos** [comment]
@gtcno It should, but it does not. I have not found out why.

**Comment by linemos** [comment]
Cannot reproduce the problem right now. Perhaps the bug was somewhere else... 🤔

**Comment by gtcno** [comment]
This is a wormhole.  But i thought this was fixed here 
https://github.com/kubernetes/kubernetes/pull/44058.

I do think we should be careful with deleting resources that we dont explicitly create.
Are we sure the deployment controller doesnt recreate a replicaset if you delete one that is created by a deployment?
And yeah we do delete the deployment. But kubernetes controllers are asyn so race conditions might occur.

**Pull request closed without merging**